### ISSUE TITLE
Warning to override LSB defaults and Failure to resolve host

### ISSUE THREAD

**Issue by xovertheyearsx** [Und3rf10w/kali-anonsurf#10]
I read an article detailing your work on [blackmoreops](https://www.blackmoreops.com/2016/10/17/browse-anonymously-in-kali-linux-with-anonsurf/) and decided to test it out on my Ubuntu 16.04.1 system. I installed `anonsurf` using `sudo bash installer.sh`. During installation, everything seemed ok until I hit a few warnings. The installation finished ok afterwards.

```
Setting up i2p (0.9.27-1~deb8u+1) ...
Synchronizing state of i2p.service with SysV init with /lib/systemd/systemd-sysv-install...
Executing /lib/systemd/systemd-sysv-install disable i2p
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
```

I then ran `anonsurf myip` to see my local ip and then ran `sudo anonsurf start` to test the app and it failed to establish a valid connection. I ran `anonsurf myip` to investigate and it dished out an error stating it failed to resolve a host.

```
$ sudo anonsurf start
 * killing dangerous applications
 * cleaning some dangerous cache elements
[ i ] Stopping IPv6 services:


[ i ] Starting anonymous mode:

 * Saved iptables rules

 * Modified resolv.conf to use Tor and Private Internet Access DNS
 * All traffic was redirected throught Tor

[ i ] You are under AnonSurf tunnel


$ anonsurf myip

My ip is:

curl: (6) Could not resolve host: myexternalip.com


----------------------------------------------------------------------
```

I had to modify the `/etc/resolve.conf` and manually add the required nameservers and then ran `anonsurf` again with success. It seems to be working otherwise. It seems to be because of Ubuntu using the `resolver(8)` to auto-generate the `/etc/resolv.conf` file

```
$ cat /etc/resolv.conf
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
nameserver 127.0.1.1
```

I'll do a little more digging. Let me know if anything. Thanks for the for amazing mash-up by the way!

**Comment by Und3rf10w** [comment]
Looks like the LSB errors are due to ```/etc/init.d/i2p``` possibly having its runlevels misconfigured. Can you paste in the output of ```head /etc/init.d/i2p  -n 12```? For comparison, my ```i2p``` script has the following runlevels:
```
# Default-Start: 2 3 4 5
# Default-Stop: 0 1 6
```

For some reason, it appears as though yours are configured different, but the output of the above ```head``` command can help us narrow this down.

As far as the resolver issue, that's interesting. I will look into a workaround for that on systems that have ```resolvconf``` installed. I pretty much just made this for Kali, and didn't consider how Ubuntu would be managing the files different. When I patch in a work around for this, I will also update this issue page. Thank you for the detailed write up.

**Comment by Und3rf10w** [comment]
Fixed by #12. I haven't received an update regarding the LSB things, but I couldn't reproduce that in my test environment.

Please pull the latest copy from the ```master``` branch and rerun the installer. That should fix it.

**Comment by xovertheyearsx** [comment]
> As far as the resolver issue, that's interesting. I will look into a workaround for that on systems that have resolvconf installed.

I actually fixed it after I posted this issue. Sorry it took so long to respond. If you'd like, I can see if I can grab the information you were looking for to help you implement a fix. I'm not sure I will be able to reproduce the issue though. 

> I pretty much just made this for Kali, and didn't consider how Ubuntu would be managing the files different. When I patch in a work around for this, I will also update this issue page. Thank you for the detailed write up.

Also, I would like to just say no problem. I wasn't sure if it would work. I knew that Kali and Ubuntu both use debian, so I had a hunch that the base system would be similar. I wanted to test it on a stable system that I use daily and even though I hit a couple snags, it ended up being a success. 

> Fixed by #12. I haven't received an update regarding the LSB things, but I couldn't reproduce that in my test environment.

I reviewed the code (which seems to be a mix of sh and bash syntax) and everything seemed okay. I just wasn't familiar with working with i2p or iptables manually, so I couldn't figure out if it was connected to that or not. I figured it had to do more with a config file, which it was.

Let me know if you're interested.

**Comment by Und3rf10w** [comment]
The changes I did in #12 I pushed to the latest ```master``` branch. That should have fixes for resolvconf which should have fixed this issue. If you'd like, you can pull down the latest and test it out. It should install itself over the old version.
